### ISSUE TITLE
Don't wrap multiline asserts into brackets

### ISSUE THREAD

**Issue by kalekseev** [ambv/black#177]
Don't wrap multiline asserts into bracers if one part fits into one line and another already in brackets.
 
        data = [{
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
        }]
        assert data == [{
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
            "a": "b",
        }]

after black

```
+        data = [
+            {
+                "a": "b",
+                "a": "b",
+                "a": "b",
+                "a": "b",
+                "a": "b",
+                "a": "b",
+                "a": "b",
+                "a": "b",
+            }
+        ]
+        assert (
+            data
+            == [
+                {
+                    "a": "b",
+                    "a": "b",
+                    "a": "b",
+                    "a": "b",
+                    "a": "b",
+                    "a": "b",
+                    "a": "b",
+                    "a": "b",
+                }
+            ]
+        )
```

Can't we have same formatting for asserts as we have for assignments?

        assert data == [
            {
                "a": "b",
                "a": "b",
                "a": "b",
                "a": "b",
                "a": "b",
                "a": "b",
                "a": "b",
                "a": "b",
            }
        ]

**Comment by ambv** [comment]
Yes, I agree with you. I will improve automatic parentheses handing, that will improve what's happening in this case, as well as other places currently auto-managed (if, elif, while, and so on). Then I can add it to assignments and dictionary values, too. Currently I admit it's a bit clumsy for most obvious examples (e.g. when there is just a single delimiter like the equals sign you're referring, too.).

As for unpacking multiple brackets to separate indentation level, Black will continue doing this. I don't like multiple unrelated brackets shoved together to save a line.

**Comment by dhimmel** [comment]
I think the original code, even after the extra parentheses were removed in the reformatted version, is the best. Specifically, I like:
 
```python
    assert data == [{
        "a": "b",
        "a": "b",
    }]
```

More than:

```python
    assert data == [
        {
            "a": "b",
            "a": "b",
        }
    ]
```

Putting `[{` together avoids an extra level of indentation and saves two lines.

Here's an example from [our codebase](https://github.com/manubot/manubot/pull/164), where the compact format is preferred:

```python
        args.extend([
            '--template-variables-path',
            str(manuscript_dir.joinpath('content/template-variables.json')),
        ])
```

Is there an option for this? Aware of any open issues on the matter? Or how I should title this request if opening a new issue requesting it?
